sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2694,8 +2702,7 @@ class Analyzer(
           if (missingExpr.nonEmpty) {
             extractedExprBuffer += ne
           }
-          // alias will be cleaned in the rule CleanupAliases
-          ne
+          ne.toAttribute
```

Comment:
This was found by the latest commit following the comment: https://github.com/apache/spark/pull/29585#discussion_r483444401